src/arith/rewrite_simplify.cc

```diff
@@ -353,6 +353,25 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const SubNode* op) {
         truncdiv(x + c1, c3) - truncdiv(x, c3), truncdiv(truncmod(x, c3) + c1, c3),
         CanProveGreaterEqual(x.Eval(), 0) && c1.Eval()->value >= 0 && c3.Eval()->value > 0);
 
+    TVM_TRY_REWRITE_IF(truncdiv(x * c1 + c2, c3) - truncdiv(x * c1 + c4, c3), truncdiv(c2 - c4, c3),
+                       c1.Eval()->value >= 0 && c2.Eval()->value > 0 && c3.Eval()->value > 0 &&
+                           c4.Eval()->value >= 0 && CanProveGreaterEqual(x.Eval(), 0) &&
+                           std::abs(c4.Eval()->value - c2.Eval()->value) >= c3.Eval()->value);
```

Comment:
hmm, is this always true again?

say `c1 = 0, c2 = 6, c3 = 3, c4 = 2`, `c4 - c2 = 4 > 3`
`truncdiv(x * c1 + c2, c3) - truncdiv(x * c1 + c4, c3) = truncdiv(6, 3) - truncdiv(2, 3)`
`truncdiv(c2 - c4, c3) = truncdiv(4, 3)`

Reply:
I add your case in unitest. truncdiv(6, 3)  will be simplified to 2 first, so what we catch here is '2-tdiv(2,3)'.

Reply:
truncdiv(6, y) - truncdiv(2, y) if y=3 will break the rule, remove y related rule for now.